packages/admin/admin-bundler/src/utils/config.ts

```diff
@@ -70,6 +70,21 @@ export async function getViteConfig(
     ],
   }
 
+  // Inject plugin environment variables as vite defines
+  for (const plugin of options.plugins ?? []) {
+    const normalizedName = plugin.replace(/\/admin$/, "")
+      .replace(/^@/, "")
+      .replace(/[^a-zA-Z0-9]/g, "_")
+      .toUpperCase()
+    const pluginEnv: Record<string, string | undefined> = {}
+    for (const envVar in process.env) {
+      if (envVar.startsWith(`PLUGIN_${normalizedName}_`)) {
+        pluginEnv[envVar] = process.env[envVar];
+      }
+    }
+    baseConfig.define!["process.env"] = JSON.stringify(pluginEnv);
+  }
+
```

Comment:
suggestion: Here it seems that for every plugin loop we are overriding the `process.env` in define. Also, I propose a little change to this, let me know what you think
```suggestion
const pluginEnv = Object.entries(process.env).filter(([key]) =>
    key.startsWith("PLUGIN_")
  )
  const pluginEnvMap: Record<string, string | undefined> = {}
  if (pluginEnv.length) {
    for (const plugin of options.plugins ?? []) {
      const normalizedName = plugin
        .replace(/\/admin$/, "")
        .replace(/^@/, "")
        .replace(/[^a-zA-Z0-9]/g, "_")
        .toUpperCase()
      for (const [env, value] of pluginEnv) {
        if (env.startsWith(`PLUGIN_${normalizedName}_`)) {
          pluginEnvMap[env] = value
        }
      }
    }
    baseConfig.define!["process.env"] = JSON.stringify(pluginEnvMap)
  }
```

Reply:
Good catch, I was lazy and only tested it with one, don't know how I didn't see that just by looking at it tho 

This got me thinking: for some reason in my head I thought each plugin had its own process.env, since the are actually all shared in a single environment, should we get rid of "normalizedName" and just use "PLUGIN_" as a prefix? It would make it simpler to use especially for end users.

Reply:
I think that could be a nice approach, wdyt @willbouch

Reply:
Yep! Sounds good